### DISCUSSION TITLE
Toggleable circuit states

### DISCUSSION THREAD

**Discussion by Wallabeng** [dominikamann/oekofen-pellematic-compact#28] [Ideas]
Hy!
First of all thanks alot for this custom component.
I want to discuss the idea of having the circuit states toggleable. So I can turn them on/off, depending on external factors.
Personally I would like to turn them off when I fire my wood stove in the living room and let them turn on automatically again in the night.
Specifically I'm speaking of the value mode_auto, which can be set like this:
```
<ip>:<port>/<pw>/hk<circuit>.mode_auto=<value>
```
Where value 0 is off, 1 is auto, 2 is heat and 3 is setback. (Personally I would only need 0 and 1).

Maybe editing of values can be extended to other settings. But lets start here.

**Comment by loicguiraud** [comment]
thanks a lot for this information
I thought that JSON feature was only able to read settings but it's not !
I was able to create automation using this method, i just use rest_command to call a specific url and use it in automation

**Reply by Wallabeng** [reply]
yes, would be possible with that too. I just want to have this integrated in this component, as it's more comfortable :-)

**Comment by dominikamann** [comment]
Hi @Wallabeng,
this is a great idea. It will also help me to implement a more intelligent heating control in combination with my PV surplus heating and weather data. I will have a look on it as soon as I have some free time 👍

**Comment by briceparent** [comment]
Hello!
Having that out of the box would be amazing! I had no idea the json endpoint was read+write!
Any plan to integrate this into the component? And if not, can anyone direct me in the direction of how to implement it manually? I'm a developer, but I'm quite new with HA and home automation in general, so I'm not even sure of where to start for this!
Thanks!
Brice

**Reply by dominikamann** [reply]
Hi @briceparent,
it is planned to integrate it, but I am very busy at the moment. As soon as I have time, I will do it. In the meantime, you can check https://www.home-assistant.io/integrations/rest_command/. You can easily do it manually with that.

**Reply by briceparent** [reply]
Thanks a lot!
If I manage to make this work, I'll try to share how I did it here.

**Comment by dominikamann** [comment]
It is now possible, thanks to @sweetcorreze !
See release: https://github.com/dominikamann/oekofen-pellematic-compact/releases/tag/v3.3.0
